cmd/migrate/db/mariadb.go

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+package db
+
+import (
+	"fmt"
+	"os"
+
+	"database/sql"
+
+	_ "github.com/go-sql-driver/mysql"
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/scribe-org/scribe-server/cmd/migrate/types"
+	"gopkg.in/yaml.v2"
+)
+
+// LoadConfig reads and unmarshals the YAML config file.
+func LoadConfig(path string) (*types.Config, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read config file: %v", err)
+	}
+	var config types.Config
+	if err := yaml.Unmarshal(data, &config); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal config: %v", err)
+	}
+	return &config, nil
+}
```

Comment:
Also - loading the config file with `yaml.Unmarshal` is fine with me :+1:  I would just suggest though for both `api` and `migrate` to use the same implementation, be it using `yaml.Unmarshal` or `viper` (as in [here](https://github.com/scribe-org/Scribe-Server/blob/main/main.go#L13-L35)).

I do like the use of a `Config` struct though :blush: so we should definitely keep that. `viper` appears to be able to use that too if that's the path chosen ([ref](https://github.com/spf13/viper?tab=readme-ov-file#unmarshaling)), but either implementation is fine